02/app.js

```diff
@@ -1,7 +1,32 @@
-
 /* rozwiązanie z pętlą for */
-const x = 4;
+
+const multiplier = getOneDigit('liczbę');
+
+for (let i = 1; i <= 9; i++) {
+    console.log(`${i} x ${multiplier} = ${i * multiplier}`);
+}
+
+console.log();
+
+/* rozwiązanie z pętlą while  */
+const base = getOneDigit('podstawę');
+const exponent = getOneDigit('wykładnik');
+
+let multLeft = exponent;
+let output = `${base}`;
+while(--multLeft) {
+    output += ` x ${base}` ;
+}
+output += ` = ${base ** exponent}`;
+
+console.log(output);
 
 
+function getOneDigit(purpose) {
+    let x = +prompt(`Podaj ${purpose}!`);
 
-/* rozwiązanie z pętlą while  */
\ No newline at end of file
+    while (!(x < 9) || !(x > 1)) {
```

Comment:
Nie będzie prościej `x > 9 || x < 1 `?

Reply:
uznałam że to będzie też zabezpieczenie przed NaNami, undefinedami itd :p może tak być czy to nie ma sensu?